Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,369 @@
+    print("""
+K8s Version Recency Compliance Check
+
+Usage: k8s-version-recency-check.py [-h] [-t|--token GITHUB_TOKEN] [-k|--kubeconfig PATH/TO/KUBECONFIG]
+
+The K8s version recency check returns 0 if the version of the tested cluster is still acceptable, otherwise
+it returns 1 for an out-of-date minor version, 2 for an out-of-date patch level version or 3 if the currently
+used version should be updated due to a highly critical CVE.
+
+    -t/--token GITHUB_TOKEN            - Github token used to check the Github API for the latest K8s releases
+    -k/--kubeconfig PATH/TO/KUBECONFIG - Path to the kubeconfig of the server we want to check
+    -h                                 - Output help
+    """)
+    return
+
+
+def parse_arguments(argv):
+    """Parse cli arguments from the script call
+    """
+    config = Config()
+    config_path = "./config.yaml"
+
+    try:
+        opts, args = getopt.gnu_getopt(argv, "t:k:h", ["token", "kubeconfig", "help"])
+    except getopt.GetoptError:
+        print_usage()
+        raise ConfigException
+
+    for opt in opts:
+        if opt[0] == "-h" or opt[0] == "--help":
+            print_usage()
+            raise HelpException
+        if opt[0] == "-c" or opt[0] == "--config":
+            config_path = opt[1]
+        if opt[0] == "-k" or opt[0] == "--kubeconfig":
+            config.kubeconfig = opt[1]
+
+    return config, config_path
+
+
+def setup_logging(config_log):
+
+    logging.config.dictConfig(config_log)
+    loggers = [
+        logging.getLogger(name)
+        for name in logging.root.manager.loggerDict
+        if not logging.getLogger(name).level
+    ]
+
+    for log in loggers:
+        log.setLevel(config_log['level'])
+
+
+def initialize_config(config, config_path):
+    """Initialize the configuration for the test script
+    """
+
+    config_file = None
+    try:
+        with open(config_path, "r") as fd:
+            config_file = yaml.safe_load(fd)
+    except OSError:
+        pass
+
+    if config_file:
+        # Setup logging if the config file with the relevant information could be loaded before
+        setup_logging(config_file['log'])
+    else:
+        # Otherwise, we initialize basic logging
+        logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
+
+    if not config.kubeconfig:
+        print("A kubeconfig needs to be set in order to test a k8s cluster version.")
+        raise ConfigException
+
+    return config
+
+
+class K8sVersionInfo:
+    """Class that contains a k8s version info.
+
+    Attributes:
+        major (int): Major version of the k8s version
+        minor (int): Minor version of the k8s version
+        patch (int): Patch version of the k8s version
+        date (datetime): release date of the k8s version
+    """
+    def __init__(self, major=0, minor=0, patch=0):
+        self.major = major
+        self.minor = minor
+        self.patch = patch
+
+        self.date = None
+
+    def __eq__(self, other):
```

Comment:
I meant to say that if you implement `__eq__`, you also have to implement `__ne__`, BUT apparently, that is no longer true for Python 3. Just FYI.